src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        validateUserData(user); // Validate user data before registration
+        return storageDao.add(user);
     }
+
+    private void validateUserData(User user) {
+        if (user.getLogin().length() < 6
```

Comment:
Don't create one big if with all conditions. Better to split one big if into several small ones.